main.go

```diff
@@ -338,9 +373,61 @@ func determineImageConstraints() models.ImageConstraints {
 			Version: determineNodeVersion(),
 		}
 	}
+	imageConstraints.DatabaseTypes = determineDatabaseTypes()
 	return imageConstraints
 }
 
+// determineDatabaseTypes returns a set of database types needed for tests
+func determineDatabaseTypes() map[string]struct{} {
+	databaseTypes := map[string]struct{}{}
+	if needsPostgreSQL() {
+		databaseTypes[POSTGRESQL_DB_TYPE] = struct{}{}
+	}
+	if needsMongoDB() {
+		databaseTypes[MONGO_DB_TYPE] = struct{}{}
+	}
+	return databaseTypes
+}
+
+// needsPostgreSQL returns true if tests rely on postgresql, based on these criteria:
+// -- true if a Makefile contains the text `psql`
+// -- false otherwise
+func needsPostgreSQL() bool {
+	postgresqlCheckRegexp := regexp.MustCompile(`psql`)
+	makefile, err := ioutil.ReadFile("Makefile")
+	if err != nil {
+		log.Fatal(err)
+	}
+	return postgresqlCheckRegexp.Match(makefile)
+}
+
+// needsMongoDB returns true if tests rely on mongodb, based on these criteria:
+// -- true if Makefile contains the text `MONGO_TEST_DB`
+// -- true if a file with `test` in the name contains the text `testMongoURL`
+// -- false otherwise
+func needsMongoDB() bool {
+	// check Makefile for MONGO_TEST_DB
+	mongoCheckRegexp := regexp.MustCompile(`MONGO_TEST_DB`)
```

Comment:
Same Q on checking dependencies instead

Reply:
I think we have some repos with multiple mongo connections that go "LEGACY_TEST_DB" & "SIS_TEST_DB" (or patterns like this)